tests/integration_tests/agent/test_pandas_agent.py

```diff
@@ -0,0 +1,40 @@
+import re
+
+import numpy as np
+import pytest
+from pandas import DataFrame
+from pydantic import BaseModel
+
+from langchain.agents import create_pandas_dataframe_agent
+from langchain.agents.agent import AgentExecutor
+from langchain.llms import OpenAI
+
+
+class TestData(BaseModel):
+    df: DataFrame
+    dim: int
+
+    class Config:
+        arbitrary_types_allowed = True
+
+
+@pytest.fixture(scope="function")
+def data() -> TestData:
+    dim = 4
+    random_data = np.random.rand(dim, dim)
+    df = DataFrame(random_data, columns=["name", "age", "food", "sport"])
+    return TestData(df=df, dim=4)
```

Comment:
is dim same as `len(df)`? could we just use that and not need to define a TestData class?

Reply:
sure, just updated.